go/libraries/doltcore/remotesrv/interceptors.go

```diff
@@ -66,9 +69,10 @@ func (si *ServerInterceptor) Options() []grpc.ServerOption {
 }
 
 func (si *ServerInterceptor) authenticate(ctx context.Context) error {
-	if len(si.ExpectedUserAuth.User) == 0 && len(si.ExpectedUserAuth.Password) == 0 {
+	if !si.ShouldAuth || si.Authenticator == nil {
```

Comment:
What's going on here? Shouldn't we run this code even if a static username and password aren't configured? Isn't that the point of adding the grants support?

FWIW, if there are fast path exits like this, I think we should try to not install this interceptor at all instead. So, like, instead of ShouldAuth being a thing here, we could just do:

```go
if shouldAuth {
    args = args.WithUserPasswordAuth(...)
}
```